source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -1469,18 +1471,22 @@ RtPrim RtFileIo::readPrim(std::istream& stream, const RtReadOptions* options)
             }
             else if (elem->isA<Node>())
             {
-                PvtPrim* p = readNode(elem->asA<Node>(), stage->getRootPrim(), stage, mapper);
+                PvtPrim* p = readNode(elem->asA<Node>(), stage->getPrimAtPath(parentPath), stage, mapper);
                 return p ? p->prim() : RtPrim();
             }
             else if (elem->isA<NodeGraph>())
             {
+                if (parentPrimPath.asString() != "/")
```

Comment:
If it's not a root parent prim path then we are creating a nested compound which isn't supported so throw an exception.